[Term Entry] Nmap: UDP Scan

jansoth opened the pull request:
### Description

A new contribution to the Codecademy Docs that adds a markdown file which describes the "UDP-scan" mode to the nmap section. The file is located in content/cybersecurity/concepts/nmap/terms/udp-scan.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Codecademy/docs?pullRequest=1341) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=1341) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=1341) it.</sub>

SSwiniarski commented on content/cybersecurity/concepts/nmap/terms/udp-scan/udp-scan.md:
```diff
@@ -0,0 +1,58 @@
+  - 'paths/scan-systems-with-nmap'
+---
+
+An **UDP Scan** scans (only) the UDP Ports of a target, while the [regular scan](https://www.codecademy.com/resources/docs/cybersecurity/nmap/nmap-scan) only scans the TCP Ports. It is performed by using the -sU option and can be combined with any TCP scan type, for example SYN scan (-sS) to scan UDP and TCP Ports in the same run. UDP Scan is normally slower and more difficult than a TCP scan.
```
```suggestion
An **UDP Scan** scans (only) the UDP Ports of a target, while the [regular scan](https://www.codecademy.com/resources/docs/cybersecurity/nmap/nmap-scan) only scans the TCP Ports. It is performed by using the `-sU` option and can be combined with any TCP scan type, for example SYN scan (`-sS`) to scan UDP and TCP Ports in the same run. UDP Scan is normally slower and more difficult than a TCP scan.
```

jansoth replied:
I have a question about the md file: the metadata at the beginning isn't displayed correctly when opening https://github.com/jansoth/CodecademyDocs/blob/NmapUDPScan/content/cybersecurity/concepts/nmap/terms/udp-scan/udp-scan.md (as shown in the image).
I can't find the mistake when comparing the source to for example the 'agressive scan' source in the md file

![grafik](https://user-images.githubusercontent.com/25724976/208101233-60ad3871-b0b3-4767-b34c-09cad569661b.png)

SSwiniarski reviewed (changes requested):
@jansoth I did a first review and only had one comment.

SSwiniarski commented on content/cybersecurity/concepts/nmap/terms/udp-scan/udp-scan.md:
```diff
@@ -0,0 +1,58 @@
+---
+Title: 'UDP Scan' 
+Description: 'Scans (only) the UDP Ports of a target, which aren't scanned in a regular scan. Can be combined with any TCP scan' 
```
```suggestion
Description: 'Scans (only) the UDP Ports of a target, which are not scanned in a regular scan. Can be combined with any TCP scan' 
```

SSwiniarski replied:
The Metadata doesn't like the apostrophe in the description.

jansoth replied:
Thank you very much! That would have taken forever for me to find out

SSwiniarski reviewed (changes requested):
@jansoth, I have another comment. The apostrophe in the description is throwing off the Metadata.

SSwiniarski reviewed (approved):
@jansoth looks good for a second review.

jansoth commented:
I have another Question: the format:verify test fails. How can i make it pass?

Dusch4593 commented:
> I have another Question: the format:verify test fails. How can i make it pass?

Hi @jansoth! 👋🏻 Most likely, the format:verify test is not passing because of trailing spaces in the first 14 lines of the `upd-scan.md` file:

<img width="1112" alt="Screen Shot 2022-12-20 at 7 23 56 PM" src="https://user-images.githubusercontent.com/23144845/208792343-9dd9a7f5-5498-49d3-9a08-3c0b98234649.png">

Let's delete those extra trailing spaces and re-push to see if the check passes 😄

jansoth commented:
I removed the trailing whitespaces. @Dusch4593 could you review please?

Dusch4593 commented:
@jansoth All of the checks are now passing! 🎉

However we seem to have a new issue here. How come there are now 92 other files attached to this PR?

jansoth commented:
@Dusch4593 I fixed the problem. Now everything should be fine. Maybe it was because i rebased the Nmap Branch to main? I still have problems understanding the workflow when reviewing PRs. Could you give me a brief explanation? 

I committed the suggested changes from SSwiniarski. What would have been the next step? Pull from origin to my local Nmap branch then fix the whitespace issue and then push again to origin?

Dusch4593 commented:
> @Dusch4593 I fixed the problem. Now everything should be fine. Maybe it was because i rebased the Nmap Branch to main? I still have problems understanding the workflow when reviewing PRs. Could you give me a brief explanation?
> 
> I committed the suggested changes from SSwiniarski. What would have been the next step? Pull from origin to my local Nmap branch then fix the whitespace issue and then push again to origin?

@jansoth Yes, that sounds about right. If there was work done on the remote `origin` version of `NmapUDPScan` (directly on GitHub), it needs to be pulled down to continue work on the local version of the branch (i.e., git pull origin NmapUDPScan).

I'm afraid I'm not too familiar with rebasing in Git, but that seems plausible as to why so many files got involved. 😅

Dusch4593 commented:
@jansoth I'm gonna take a look at your contribution for review 2️⃣  😄

Dusch4593 commented on content/cybersecurity/concepts/nmap/terms/udp-scan/udp-scan.md:
````diff
@@ -0,0 +1,58 @@
+nmap -sU <target>
+```
+
+The above conducts an UDP scan on `<target>`.
````
```suggestion
The `-sU` option is used to perform a UDP scan on a `<target>`. It can be combined with any TCP scan type. For example, a stealth (SYN) scan (`-sS`) checks for UDP and TCP ports in the same run.
```

Dusch4593 commented on content/cybersecurity/concepts/nmap/terms/udp-scan/udp-scan.md:
```diff
@@ -0,0 +1,58 @@
+
+## Example
+
+The following example runs an UDP scan on the site [scanme.nmap.org](http://scanme.nmap.org/):
```
```suggestion
The following example runs a UDP scan on the site [scanme.nmap.org](http://scanme.nmap.org/):
```

Dusch4593 commented on content/cybersecurity/concepts/nmap/terms/udp-scan/udp-scan.md:
```diff
@@ -0,0 +1,58 @@
+  - 'paths/scan-systems-with-nmap'
+---
+
+An **UDP Scan** scans (only) the UDP Ports of a target, while the [regular scan](https://www.codecademy.com/resources/docs/cybersecurity/nmap/nmap-scan) only scans the TCP Ports. It is performed by using the `-sU` option and can be combined with any TCP scan type, for example SYN scan (`-sS`) to scan UDP and TCP Ports in the same run. UDP Scan is normally slower and more difficult than a TCP scan ...
```
```suggestion
A **user diagram protocol (UDP) scan** checks for any UDP ports that are deployed on a target. Conversely, the [regular scan](https://www.codecademy.com/resources/docs/cybersecurity/nmap/nmap-scan) only scans the TCP ports. UDP scans are normally slower and more difficult than TCP scans.
```

I think the second sentence makes more sense somewhere in the "Syntax" section below.

Dusch4593 commented on content/cybersecurity/concepts/nmap/terms/udp-scan/udp-scan.md:
```diff
@@ -0,0 +1,58 @@
+---
+Title: 'UDP Scan'
+Description: 'Scans (only) the UDP Ports of a target, which are not scanned in a regular scan. Can be combined with any TCP scan'
```
```suggestion
Description: 'Checks for any UDP ports deployed on a target and can be combined with any TCP scan.'
```

The `Description` should be end with a period and (preferably) be one sentence.

Dusch4593 commented on content/cybersecurity/concepts/nmap/terms/udp-scan/udp-scan.md:
```diff
@@ -0,0 +1,58 @@
+Title: 'UDP Scan'
+Description: 'Scans (only) the UDP Ports of a target, which are not scanned in a regular scan. Can be combined with any TCP scan'
+Subjects:
+  - 'Computer Science'
+  - 'Networking'
```
```suggestion
  - 'Computer Science'
  - 'Cybersecurity'
  - 'Networking'
```

Dusch4593 reviewed (changes requested):
@jansoth Finished with taking a look and left you some comments/suggestions. 😄

jansoth commented:
Thank you for your comments!

Dusch4593 reviewed (approved):
@jansoth LGTM! 👍🏻

Dusch4593 commented:
Congrats and thank you so much for contributing to Codecademy Docs, @jansoth!

The entry should now be live: https://www.codecademy.com/resources/docs/cybersecurity/nmap/udp-scan

<img width="827" alt="Screen Shot 2022-12-28 at 1 00 52 PM" src="https://user-images.githubusercontent.com/23144845/209853646-4cb35546-5824-40c4-855f-d19c17d351c2.png">

Looking forward to your next Docs contribution! 😄

jansoth commented:
Thank you! :)
